subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -344,7 +341,7 @@ static int __wifi_args_to_params(size_t argc, char *argv[],
 	}
 
 	/* PSK (optional) */
-	if (idx < argc) {
+	if (argv[idx]) {
```

Comment:
If `idx` is more than argc, then this could point to an arbitrary memory and this check might pass.